lib/fs.js

```diff
@@ -243,10 +247,18 @@ fs.readFile = function(path, options, callback_) {
     return;
 
   var context = new ReadFileContext(callback, encoding);
+  context.isUserFd = isFd(path); // file descriptor ownership
   var req = new FSReqWrap();
   req.context = context;
   req.oncomplete = readFileAfterOpen;
 
+  if (context.isUserFd) {
+    process.nextTick(function() {
+      req.oncomplete.call(req, null, path);
```

Comment:
the default context is the object to the left of the `.` so I think @thefourtheye is right in this case.